### ISSUE TITLE
Implement RabbitMQ Client 7.0

### ISSUE THREAD

**Issue by ArieGato** [ArieGato/serilog-sinks-rabbitmq#156]
Investigate consequences of all breaking changes.

Can we support both versions? Maybe create two nuget packages.

**Comment by sungam3r** [comment]
> Maybe create two nuget packages.

Not two packages with different names, but two different versions - v7 and v8. You already bumped serilog to v3 that is a breaking change. I suggest to accumulate all other changes in v7 except bumping rabbitmq to v7, release v7 of sink and then publish v8 of sink with rabbitmq v7.

**Comment by ArieGato** [comment]
This is definitly not going to be part of v7. It's more of a discussion about how we want to handle this. 

At the moment RabbitMQClient 7 has an alpha status.

The question is whether we want to stop supporting RabbitMQClient 6.x in v8.

**Comment by sungam3r** [comment]
It depends of when rabbitmq goes from alpha to stable. It can be in alpha for years. V8 of this sink may have some unrelated to rabbitmq 7 breaking changes. Who knows.

**Comment by sungam3r** [comment]
https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/1413
https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/1056

**Comment by lukebakken** [comment]
Hello! I am the primary maintainer of the RabbitMQ .NET client. If you would like to speed along the release of version 7, testing alpha releases is a great way to help out -

https://www.nuget.org/packages/RabbitMQ.Client/7.0.0-alpha.6

I really would like to get everything wrapped up this month 🤞

**Comment by ArieGato** [comment]
@lukebakken Is there a migration guid?

**Comment by lukebakken** [comment]
@ArieGato no, there is no guide at this point in time. The API remains very similar to before, and the test suites are very thorough, so you should be able to figure out the necessary changes. If you need assistance, the best place to ask is here:

https://github.com/rabbitmq/rabbitmq-dotnet-client/discussions

One "big" change is the rename of `IModel` to `IChannel`.

**Comment by ArieGato** [comment]
Hi @lukebakken I implemented the RabbitMQ client 7. Do you have an estimate when the final release will be. I want to start scheduling a breaking version myself and would like to include the new client.
